storage/badger/chunk_data_pack_test.go

```diff
@@ -15,35 +18,92 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestChunkDataPack(t *testing.T) {
+// TestChunkDataPacks_Store evaluates correct storage and retrieval of chunk data packs in the storage.
+// It also evaluates that re-inserting is idempotent.
+func TestChunkDataPacks_Store(t *testing.T) {
+	WithChunkDataPacks(t, 100, func(t *testing.T, chunkDataPacks []*flow.ChunkDataPack, chunkDataPackStore *badgerstorage.ChunkDataPacks, _ *badger.DB) {
+		wg := sync.WaitGroup{}
+		wg.Add(len(chunkDataPacks))
+		for _, chunkDataPack := range chunkDataPacks {
+			go func(cdp flow.ChunkDataPack) {
+				err := chunkDataPackStore.Store(&cdp)
+				require.NoError(t, err)
+
+				wg.Done()
+			}(*chunkDataPack)
+		}
+
+		unittest.RequireReturnsBefore(t, wg.Wait, 1*time.Second, "could not store chunk data packs on time")
+
+		// re-insert - should be idempotent
+		for _, chunkDataPack := range chunkDataPacks {
+			err := chunkDataPackStore.Store(chunkDataPack)
+			require.NoError(t, err)
+		}
+	})
+}
+
+// TestChunkDataPack_BatchStore evaluates correct batch storage and retrieval of chunk data packs in the storage.
+func TestChunkDataPacks_BatchStore(t *testing.T) {
+	WithChunkDataPacks(t, 100, func(t *testing.T, chunkDataPacks []*flow.ChunkDataPack, chunkDataPackStore *badgerstorage.ChunkDataPacks, db *badger.DB) {
+		batch := badgerstorage.NewBatch(db)
+
+		wg := sync.WaitGroup{}
+		wg.Add(len(chunkDataPacks))
+		for _, chunkDataPack := range chunkDataPacks {
+			go func(cdp flow.ChunkDataPack) {
+				err := chunkDataPackStore.BatchStore(&cdp, batch)
+				require.NoError(t, err)
+
+				wg.Done()
+			}(*chunkDataPack)
+		}
+
+		unittest.RequireReturnsBefore(t, wg.Wait, 1*time.Second, "could not store chunk data packs on time")
+
+		err := batch.Flush()
+		require.NoError(t, err)
+	})
+}
+
+// TestChunkDataPacks_MissingItem evaluates querying a missing item returns a storage.ErrNotFound error.
+func TestChunkDataPacks_MissingItem(t *testing.T) {
 	unittest.RunWithBadgerDB(t, func(db *badger.DB) {
 		transactions := badgerstorage.NewTransactions(&metrics.NoopCollector{}, db)
 		collections := badgerstorage.NewCollections(db, transactions)
-		store := badgerstorage.NewChunkDataPacks(&metrics.NoopCollector{}, db, collections, 100)
+		store := badgerstorage.NewChunkDataPacks(&metrics.NoopCollector{}, db, collections, 1)
```

Comment:
One issue that caused this breaking change to go silent was the cache size of `100`, which would avoid any read from storage during the tests, as all would be addressed by the cache. We set the cache size to 1 to make sure storage retrieval is tested properly.